src/scenicplus/data_wrangling/cistarget_wrangling.py

```diff
@@ -120,14 +124,24 @@ def _get_cistromes(
                 target_regions_motif_extended.update(motif_hits[motif])
             else:
                 raise ValueError(f"Motif enrichment table and motif hits don't match for the TF: {tf_name}")
+
         cistromes.append(
             Cistrome(
                 tf_name = tf_name,
                 motifs = set(motifs_annotated_to_tf),
-                target_regions = target_regions_motif_extended & scplus_regions,
+                target_regions =  _overlap_if_necessary(target_regions_motif_extended,  scplus_regions, subset_regions),
                 extended = True))
     return cistromes
 
+def _overlap_if_necessary(s_query_regions, test_regions, regions_to_overlap):
```

Comment:
If I'm not mistaken this function returns either a set (else block) or a pyranges object (if block), while the Cistrome class expects a set for the `target_regions` parameter.

Also I would run this function not while providing the arguments to clearly seperate these two things (i.e. 1. initiating the class and 2. putting the subset region in the same coordinate system as SCENIC+).